odxtools/preconditionstateref.py

```diff
@@ -0,0 +1,142 @@
+            doc_frags: List[OdxDocFragment]) -> "PreConditionStateRef":
+        kwargs = dataclass_fields_asdict(OdxLinkRef.from_et(et_element, doc_frags))
+
+        value = et_element.findtext("VALUE")
+
+        in_param_if_snref = None
+        if (in_param_if_snref_elem := et_element.find("IN-PARAM-IF-SNREF")) is not None:
+            in_param_if_snref = odxrequire(in_param_if_snref_elem.get("SHORT-NAME"))
+
+        in_param_if_snpathref = None
+        if (in_param_if_snpathref_elem := et_element.find("IN-PARAM-IF-SNPATHREF")) is not None:
+            in_param_if_snpathref = odxrequire(in_param_if_snpathref_elem.get("SHORT-NAME-PATH"))
+
+        return PreConditionStateRef(
+            value=value,
+            in_param_if_snref=in_param_if_snref,
+            in_param_if_snpathref=in_param_if_snpathref,
+            **kwargs)
+
+    def __post_init__(self) -> None:
+        if self.value is not None:
+            odxassert(self.in_param_if_snref is not None or self.in_param_if_snref is not None,
+                      "If VALUE is specified, a parameter must be referenced")
+
+    def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
+        return {}
+
+    def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
+        self._state = odxlinks.resolve(self, State)
+
+    def _resolve_snrefs(self, context: SnRefContext) -> None:
+        pass
+
+    def applies(self, state_machine: "StateMachine", params: List[Parameter],
+                param_value_dict: ParameterValueDict) -> bool:
+        """Given a state machine, evaluate whether the precondition is fulfilled or not
+
+        Note that the specification is unclear about what the
+        parameters are: It says "The optional VALUE together with the
+        also optional IN-PARAM-IF snref at STATE-TRANSITION-REF and
+        PRE-CONDITION-STATE-REF can be used if the STATE-TRANSITIONs
+        and pre-condition STATEs are dependent on the values of the
+        referenced PARAMs.", but it does not specify what the
+        "referenced PARAMs" are. For the state transition refs, let's
+        assume that they are the parameters of a positive response
+        received from the ECU, whilst we assume that they are the
+        parameters of the request for PRE-CONDITION-STATE-REFs.
+        """
+
+        if state_machine.active_state != self.state:
+            # if the active state of the state machine is not the
+            # specified one, the precondition does not apply
+            return False
+
+        if self.in_param_if_snref is None and self.in_param_if_snpathref is None:
+            # no parameter given -> only the specified state is relevant
+            return True
+
+        param, param_value = \
+            _resolve_in_param(self.in_param_if_snref, self.in_param_if_snpathref, params, param_value_dict)
+
+        if param is None:
+            # The referenced parameter does not exist. TODO: is this an error?
```

Comment:
No reason for it to be
```suggestion
            # The referenced parameter does not exist.
```